LibVLCSharp/Shared/MediaSource.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace LibVLCSharp.Shared
+{
+    public class MediaSource : IMediaSource
+    {
+        static MediaSource()
+        {
+            Core.Initialize();
+        }
+
+        protected MediaSource(params string[] cliOptions)
+        {
+            LibVLC = new LibVLC(cliOptions);
+            MediaPlayer = new MediaPlayer(LibVLC);
+        }
+
+        protected MediaSource(string uri) : this()
+        {
+            MediaPlayer.Media = new Media(LibVLC, uri, Media.FromType.FromLocation);
+        }
+
+        ~MediaSource()
+        {
+            Dispose();
+        }
+
+        public MediaPlayer MediaPlayer { get; }
+        public LibVLC LibVLC { get; }
+
+#if WINDOWS
+        public IntPtr Hwnd
+        {
+            set { MediaPlayer.Hwnd = value; }
+        }
+#elif ANDROID
+        public void SetAndroidContext(IntPtr aWindow)
+        {
+            MediaPlayer.SetAndroidContext(aWindow);
+        }
+#elif COCOA
+        public IntPtr NsObject
+        {
+            set { MediaPlayer.NsObject = value; }
+        }
+#endif
+
+        public static Task<MediaSource> CreateAsync(params string[] cliOptions)
+        {
+            return Task.Run(() => new MediaSource(cliOptions));
+        }
+
+        public static Task<MediaSource> CreateFromUriAsync(string uri)
```

Comment:
Maybe add an extra ctor with both `cliOptions` and `uri` for lazy devs? :-P

Reply:
I'm going to update the `CreateFromUriAsync(string uri)` to `CreateFromUriAsync(string uri, param string[] cliOptions)`

I removed the signature `MediaSource.CreateFromMediaPlayer(MediaPlayer mediaPlayer)` because the MediaPlayer implements ISource. So, you can pass a MediaSource class or a MediaPlayer class to your VideoView. I don't know what is best :
- let as this, or
- remove the ISource interface from MediaSource (and the developer should pass MediaSource.MediaPlayer), or
- remove the ISource interface from the MediaPlayer and encapsulate the MediaPlayer to a class that only implements the ISource interface (in this case, we add the signature MediaSource.CreateFromMediaPlayer)